src/opensearch.ts

```diff
@@ -490,6 +490,48 @@ export const findBalanceByAddress =
     );
   };
 
+  export const findBalances =
+  (os: Client) =>
+  (
+    currencyIdentifier: string | null,
+    term: string,
+  ): TaskEither<ApplicationError, Result<Array<Balance>>> => {
+    return pipe(
+      findSnapshot(os)("latest", currencyIdentifier),
```

Comment:
That part gets executed even if `isOrdinal(term)` is `true` which is wrong because you should fetch latest snapshot (to figure out its ordinal) only if the caller didn't specify the ordinal.

The logic should be as following:

```
If isOrdinal(term) is true
  getByFieldQuery(term) // Fetch balances for specific ordinal
else 
  latestOrdinal = findSnapshot('latest').ordinal
  getByFieldQuery(latestOrdinal) // Fetch balances for latest ordinal

```

Reply:
Makes sense, thank you for the feedback. I modified the function accordingly